### PR TITLE
Feature/native disconnect log

### PR DESCRIPTION
**Pull request by improbable-danny**
Log for if native clients are dropping connections

### CODE REVIEW

File: Game/Source/GDKTestGyms/BenchmarkGymGameMode.cpp [hunk 1]
```diff
@@ -89,6 +90,15 @@ void ABenchmarkGymGameMode::Tick(float DeltaSeconds)
 		const AActor* SpawnPoint = SpawnPoints[SpawnPointIndex];
 		SpawnNPC(SpawnPoint->GetActorLocation());
 	}
+
```
Comment (m-samiec) [inline Game/Source/GDKTestGyms/BenchmarkGymGameMode.cpp]: Move both this and the above check into a single HasAuthority() check

File: Game/Source/GDKTestGyms/BenchmarkGymGameMode.cpp [hunk 2]
```diff
@@ -89,6 +90,15 @@ void ABenchmarkGymGameMode::Tick(float DeltaSeconds)
 		const AActor* SpawnPoint = SpawnPoints[SpawnPointIndex];
 		SpawnNPC(SpawnPoint->GetActorLocation());
 	}
+	
+	if (HasAuthority() && SecondsTillPlayerCheck > 0.0f)
+	{
+		SecondsTillPlayerCheck -= DeltaSeconds;
+		if (SecondsTillPlayerCheck <= 0.0f && GetNumPlayers() != TotalPlayers)
```
Comment (m-samiec) [inline Game/Source/GDKTestGyms/BenchmarkGymGameMode.cpp]: Logic reads a little strange - perhaps TotalPlayers should be TotalExpectedPlayers?

File: Game/Source/GDKTestGyms/BenchmarkGymGameMode.cpp [hunk 3]
```diff
@@ -89,6 +90,15 @@ void ABenchmarkGymGameMode::Tick(float DeltaSeconds)
 		const AActor* SpawnPoint = SpawnPoints[SpawnPointIndex];
 		SpawnNPC(SpawnPoint->GetActorLocation());
 	}
+	
+	if (HasAuthority() && SecondsTillPlayerCheck > 0.0f)
```
Comment (m-samiec) [inline Game/Source/GDKTestGyms/BenchmarkGymGameMode.cpp]: Might be easier just to do this with a Timer callback

File: Game/Source/GDKTestGyms/BenchmarkGymGameMode.cpp [hunk 4]
```diff
@@ -48,7 +48,24 @@ ABenchmarkGymGameMode::ABenchmarkGymGameMode()
 	RNG.Initialize(123456); // Ensure we can do deterministic runs
 }
 
-void ABenchmarkGymGameMode::CheckInitCustomSpawning()
+void ABenchmarkGymGameMode::BeginPlay()
+{	
+	FTimerHandle Timer;
+	GetWorldTimerManager().SetTimer(Timer, this, &ABenchmarkGymGameMode::CheckConnections, 15.0f*60.0f, false, 0.0f);
```
Comment (improbable-danny) [inline Game/Source/GDKTestGyms/BenchmarkGymGameMode.cpp]: Not checking authority here because of the known issue

**Pull request merged**